src/main/java/com/ericsson/ei/frontend/utils/BackEndInstancesUtils.java

```diff
@@ -64,6 +66,53 @@
     private long nextTimeToParse = 0;
     private boolean savedSinceLastParsing = false;
 
+    /**
+     * Checks that all required keys exist in the incoming json object.
+     *
+     * @param JsonObject instance
+     * @return boolean
+     * */
+    public boolean hasRequiredJsonData(JsonObject instance) {
+        boolean hasNullValues = false;
+
+        // ensure required keys are present
+        if (instance.has(HOST) && instance.has(PORT) && instance.has(NAME)
+                && instance.has(CONTEXT_PATH) && instance.has(HTTPS)
+                && instance.has(DEFAULT)) {
+
+            // ensure JSON values are not null
+            for (Map.Entry<String, JsonElement> e : instance.entrySet()) {
```

Comment:
If you feel like you should comment logic like this, i think you should think about breaking that logic out into a method with a self explainable method name instead.

Reply:
Moved to a separate method.